libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1050,7 +1050,9 @@ object MapboxRouteLineUtils {
         options.routeLayerProvider.buildWayPointLayer(
             style,
             options.originIcon,
-            options.destinationIcon
+            options.destinationIcon,
+            options.waypointLayerIconOffset,
+            options.waypointLayerIconAnchor
```

Comment:
There are numerous layer related options in the maps SDK.  Every option we add is an option we have to support with SEMVER implications etc.  I think we should only add options if/when they are requested.

Reply:
I think the fact that customization for `iconOffset` and `iconAnchor` was brought up by a customer is a reason good enough to only add support for other icon properties and not other layer related options supported by Maps SDK, which can be handled on demand. Also, since we use a simple builder pattern, I am not too worried about SEMVER implications. Adding options to allow for configuration of some icon properties vs not doing it for others, to me looks like the API is incomplete. I would consider doing it for all the properties that makes sense; probably we can leave `iconSize` and `iconImage` because a developer would need to understand `expressions` for these 2 properties and I think they would just want Mapbox to handle it and keep it simple for themselves.

Reply:
There are already an overwhelming number of options available. Rather than bloat the SDK with arbitrary options we should add options according to customer needs.  After significant usage only one customer asked for these options and they are being added so the process works.  Guessing what future options might be used is a mistake.

Reply:
I beg to differ and don't think it's a mistake. Given the complexity of this feature and product itself it's obvious that there would be vast number of options available for a customer to customize. Having it available, now that we see a request only adds to the credibility of our product. Opinions aside I still stand by my comment that we should add these options as well. Feel free to choose the best approach 😄

Reply:
FYI iconIgnorePlacement, iconAllowOverlap and iconKeepUpright already default to true and have for some time.